src/nbl/asset/utils/waveContext.h

```diff
@@ -64,6 +68,7 @@ struct preprocessing_hooks final : public boost::wave::context_policies::default
         typename ContextT::token_type const& act_token
     )
     {
+        hash_token_occurences++;
```

Comment:
cool but you also need to count them on includes, ifdefs, etc.

Reply:
In boost wave `found_directive` hook triggers on every preprocessor directive, so i will count it there instead
```
template <typename ContextT, typename TokenT>
bool found_directive(ContextT const& ctx, TokenT const& directive)
{
    hash_token_occurences++;
    return false;
}
```